src/stencil.cpp

```diff
@@ -304,87 +303,108 @@ bool Stencil::traverse_dag(DAGNode* node)
     }
 
     // execute this node and return a future
-    //if (thisIndex.x == 0 && thisIndex.y == 0)
-    DEBUG_PRINT("(%i, %i)> Send ghosts called for %i\n", thisIndex.x, thisIndex.y, kernel_node->node_id);
-    send_ghost_data(kernel_node);
+    // if (thisIndex.x == 0 && thisIndex.y == 0)
+    bool ghost_exchange_needed = false;
+
+    for (int i = 0; i < kernel_node->inputs.size(); i++)
+    {
+        int ghost_depth = ghost_info[kernel_node->inputs[i]];
```

Comment:
This isn't quite right yet. The `kernel_node->inputs` array is the array of input array names passed to the kernel. Whereas the key to the `ghost_info` map is the index of the argument. So for this example,

```
@kernel
def foo(a, b):
    # ghost_info would be something like {0: 1, 1: 1}

x = create_array() # say x is assigned name 4
y = create_array() # y is assigned name 5

foo(x, y) # kernel_node->inputs = [4, 5]
```

So this should simply be `ghost_info[i]`